crates/common/src/config.rs

```diff
@@ -96,54 +100,59 @@ impl Config {
     }
 }
 
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
 pub struct IngotMetadata {
     pub name: Option<SmolStr>,
     pub version: Option<Version>,
 }
 
-#[derive(Debug, Clone)]
-pub enum DependencyDescription {
-    Path(Utf8PathBuf),
-    PathWithArguments {
-        path: Utf8PathBuf,
-        arguments: IngotArguments,
-    },
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct DependencyDescription {
+    pub url: Url,
+    pub arguments: Option<IngotArguments>,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct Dependency {
```

Comment:
It could be really nice if `Dependency` is just a flat thing, something like:

```rs
Dependency {
  alias: SmolStr,
  location: Either<Url, Utf8PathBuf>,
  arguments: Option<IngotArguments>
}